utils/orchestrator/rancherutil.py

```diff
@@ -0,0 +1,102 @@
+import requests
+from rancher_metadata import MetadataAPI
+from .baseutil import BaseUtil
+
+from utils.singleton import Singleton
+
+
+class RancherUtil(BaseUtil):
+    __metaclass__ = Singleton
+
+    METADATA_URL_AUTHORITY = "http://rancher-metadata/"
+    METADATA_VERSION = "2016-07-29"
+    METADATA_URL = "{}/{}".format(METADATA_URL_AUTHORITY, METADATA_VERSION)
+
+    CONTAINER_NAME_LABEL = "io.rancher.container.name"
+    CONTAINER_IP_LABEL = "io.rancher.container.ip"
+    STACK_NAME_LABEL = "io.rancher.stack.name"
+    SERVICE_NAME_LABEL = "io.rancher.stack_service.name"
+
+    _is_rancher = None
+
+    def __init__(self):
+        BaseUtil.__init__(self)
+        self.needs_inspect_config = True
+        self.needs_inspect_labels = True
+
+        self.api = MetadataAPI(api_url=RancherUtil.METADATA_URL)
+
+    @staticmethod
+    def is_detected():
+        return RancherUtil.is_rancher()
+
+    def _get_cacheable_tags(self, cid=None, co=None):
```

Comment:
also maybe make the default value for `co` `{}`? Otherwise `co.get` will error out